core/cat/looking_glass/stray_cat.py

```diff
@@ -156,6 +157,40 @@ def send_error(self, error: Union[str, Exception]):
 
         self.__send_ws_json(error_message)
 
+    def recall(
+            self,
+            query: str | List[float],
+            k: int | None = 5,
+            threshold: int | None = None,
+            metadata: dict | None = None,
+            collections: Iterable[MEMORY_COLLECTION] | None = None,
+            override_working_memory: bool = False
+            ):
+
+        if isinstance(query, str):
+            query = self.embedder.embed_query(query)
+
+        if collections:
+            memory_collections = get_args(MEMORY_COLLECTION)
+            collections_not_allowed = [c for c in collections if c not in memory_collections]
+            if any(collections_not_allowed):
+                raise ValueError(
+                    f"Collections not allowed: {', '.join(collections_not_allowed)}. Collections should be any of {', '.join(memory_collections)}")
+        else:
+            collections = self.memory.vectors.collections.keys()
+
+        for collection in collections:
+            vector_memory = getattr(self.memory.vectors, collection)
+            if k:
+                memories = vector_memory.recall_memories_from_embedding(
+                    query, metadata, k, threshold
+                )
+                if override_working_memory:
+                    memory_key = f"{collection}_memories"
+                    setattr(self.working_memory, memory_key, memories)
+            else:
+                memories = vector_memory.get_all_points()
```

Comment:
Yet the method as no return value because I wanted to ask how you intended to merge the memories of the different collections.
E.g. should we return a `dict` like
```python
memories = {
    "episodic_memories": # results of the recall from episodic,
   "declarative_memories": # results of the recall from declarative
}
```
similar to what we save in the working memory? Or maybe, do we want to have a rerank strategy and merge everything in a single collection?
An hybrid (and dirty?) solution could be to have an additional parameter to choose between reranking and separate results

Reply:
- if `override_working_memory == True`, just override what is contained in working memory
- if `override_working_memory == False`, do not override working memory
- in both cases, return the dictionary you prototyped above, with contents in the same format as we do for working memory

Reply:
Additional improvements if you get the time:
- method `recall_relevant_memories_to_working_memory` uses internally `recall` with `override_working_memory == True` and filters as defined by hooks
- add tests for the method in `tests/looking_glass/test_stray_cat.py` talking to the stray without the endpoints (example [here](https://github.com/cheshire-cat-ai/core/blob/31cc1ee4456c9a54cefe3c9812d433dd77f4206c/core/tests/looking_glass/test_stray_cat.py#L10))